service_analysis/app/repositories/dao/SampleObjectDao.scala

```diff
@@ -0,0 +1,67 @@
+package repositories.dao
+
+import com.google.inject.{Inject, Singleton}
+import models.SampleObject
+import no.uio.musit.MusitResults.{MusitDbError, MusitResult, MusitSuccess}
+import no.uio.musit.models.ObjectUUID
+import play.api.Logger
+import play.api.db.slick.DatabaseConfigProvider
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+@Singleton
+class SampleObjectDao @Inject() (
+    val dbConfigProvider: DatabaseConfigProvider
+) extends Tables {
+
+  val logger = Logger(classOf[SampleObjectDao])
+
+  import driver.api._
+
+  def insert(so: SampleObject): Future[MusitResult[ObjectUUID]] = {
+    val soTuple = asSampleObjectTuple(so)
+    val action = sampleObjTable += soTuple
+
+    db.run(action.transactionally).map(_ => MusitSuccess(soTuple._1)).recover {
+      case NonFatal(ex) =>
+        val msg = s"An unexpected error occurred inserting a sample object"
+        logger.error(msg, ex)
+        MusitDbError(msg, Option(ex))
+    }
+  }
+
+  def update(so: SampleObject): Future[MusitResult[Int]] = {
+    val a = sampleObjTable.filter(_.id === so.objectId).update(asSampleObjectTuple(so))
+
+    db.run(a.transactionally).map {
+      case res: Int if res == 1 => MusitSuccess(res)
+      case res: Int if 1 > res => MusitDbError("Nothing was updated")
+      case res: Int if 1 < res => MusitDbError(s"Too many rows were updated: $res")
```

Comment:
In the case that we have to many rows updated, should we do a rollback? I find it strange that we're not doing it since we have done something and send an error to the caller.